docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+title : Developer Guide
+---
+<style>
+.alert {
+    position:relative;
+    padding:.75rem 1.25rem;
+    margin-bottom:1rem;
+    border:1px solid transparent;
+    order-radius:.25rem
+}
+.alert-primary {
+    color:#073984;
+    background-image:linear-gradient(180deg, #d6e6ff, #cfe2ff);
+    border-color:#bbd6fe
+}
+.alert-warning {
+	color: #856404;
+	background-image: linear-gradient(180deg, #fff5d5, #fff3cd);
+	border-color: #ffeeba
+}
+</style>
+* Table of Contents
+{:toc}
 
-## Design & implementation
+## How To Use This Guide
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+This **Developer Guide** aims to get developers familiarised with the design and implementation of **CLIrcuit Assistant**. The following table indicates the symbols used to aid the understanding of the guide. This guide also assumes that the reader has basic understanding of *UML Diagrams*. [To access the **User Guide** instead, click here.](UserGuide.md)
 
+| Symbol/Format | Meaning |
+|:---------------:|:--------|
+|:information_source:|An informational source.|
+|:exclamation:|A warning.|
+|**Bolded**|A keyword.|
+|*Italicised*|Technical word.|
+|[Hyperlinked](#)|Leads to the appropriate section.|
+|`Code`|Text that appears on the CLI / in code.|
 
-## Product scope
-### Target user profile
+--------------------------------------------------------------------------------------------------------------------
 
-{Describe the target user profile}
+## Setting up, getting started
+Refer to the guide [*Setting up and getting started*](SettingUp.md).
 
-### Value proposition
+--------------------------------------------------------------------------------------------------------------------
 
-{Describe the value proposition: what problem does it solve?}
+## Design Architecture (Wira) <a name="design"></a>
 
-## User Stories
+![ArchitectureDiagram](diagrams/ArchitectureDiagram.png)
+
+<small><i>Figure 1</i></small>
+
+The **Architecture Design** given above explains the high-level design of the App. Given below is a quick overview of each component.
+
+**`Duke`** is the main class of the application, and handles the app launch, initializing the appropriate classes to be used.
+
+The rest of the app consists of three components.
+
+* [**`UI`**](#ui-component): The UI of the App.
+* [**`Logic`**](#logic-component): The command executor.
+* [**`Model`**](#model-component): Holds the data of the App in memory.
+
+### UI component
+
+![UiDiagram](diagrams/UiClassDiagram.png)
+
+<small><i>Figure 2</i></small>
+
+The `Ui` component
+
+* Reads user input as lines using `readLine()`.
+* Prints every `Command` object that is parsed using `printMessage()`.
+* Prints any user input error that might occur using `showError()`.
+
+![UiSequence](diagrams/UiSequenceDiagram.png)
+
+<small><i>Figure 3</i></small>
+
+### Logic component
+
+![LogicDiagram](diagrams/LogicClassDiagram.png)
+
+<small><i>Figure 4</i></small>
```

Comment:
If there are any non-trivial attributes or methods, it would be good to include them.

Reply:
Maybe you should remove the icon for the class diagram as mentioned by the prof.
Here is the link, we can refer to it. https://github.com/nus-cs2113-AY2021S1/forum/issues/112

Reply:
@czlin98 Would that make the diagram too cluttered? We already abstracted out specific `Command` classes, so this diagram just demonstrates a high level view of the `logic` package.

@Ang-Cheng-Jun Fixed in https://github.com/AY2021S1-CS2113T-W13-3/tp/pull/119.

Thanks for the feedback!